Tests/Tests.Shared/APITests.cs

```diff
@@ -33,5 +33,56 @@ public void RealmCollectionContravariance(Type type)
         {
             Assert.That(typeof(IRealmCollection<RealmObject>).IsAssignableFrom(type));
         }
+
+        [Test]
+        public void RealmIntegerApi()
+        {
+            var five = new RealmInteger<long>(5);
+            Assert.That(five == 5);
+            Assert.That(five != 10);
+            Assert.That(five.Equals(5));
+            Assert.That(five + 5, Is.EqualTo(10));
+            Assert.That(five > 5, Is.False);
+            Assert.That(five < 5, Is.False);
+            Assert.That(five > 3);
+            Assert.That(five < 7);
+            Assert.That(five / 5, Is.EqualTo(1));
+            Assert.That(five * 5, Is.EqualTo(25));
+            Assert.That(five - 3, Is.EqualTo(2));
+
+            var six = five.Increment();
```

Comment:
It should be a method - it’s what you would call on an object’s property to increment it with one. It returns the new value because the struct is immutable.